app/controllers/course/video/submission/submissions_controller.rb

```diff
@@ -10,6 +10,10 @@ def index
     @course_students = current_course.course_users.students.order_alphabetically
   end
 
+  def show
+    @sessions = @submission.sessions.with_events_present
```

Comment:
Just a quick question, how large will this data be potentially? 

My concern is that if it is really large might we want to consider loading sessions data via AJAX instead.

Reply:
I won't know for sure without actual data. 

However, from what I can anticipate, each event roughly corresponds to about one UI action. Assuming normal usage, since this is for a single video, I don't forsee it ever being over a few thousand.

Reply:
Just a quick check, prod has around 3k sessions, with around 200 sessions with more than 100 events. The top 10 sessions with highest event counts has between 600-5200 events > <  - Will that be an issue?